src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -76,6 +76,32 @@ class TezosNodeOperator(val node: TezosRPCInterface, val network: String, batchC
   type BallotBlock = (Block, List[Voting.Ballot])
   type BakerBlock = (Block, List[Voting.BakerRolls])
 
+  /**
+    * Generic fetch for paginated data relative to a specific block.
```

Comment:
In this case, the Key will usually be BlockHash, and the Entity would be any data linked to that hash, like that block's accounts/delegates, correct? Why are we generalizing over Key?